packages/maskbook/src/utils/hooks/useQueryNavigatorPermission.ts

```diff
@@ -34,26 +34,20 @@ export function useQueryNavigatorPermission(needRequest: boolean, name: Permissi
                 .then((p) => {
                     permissionStatus = p
                     permissionStatus.addEventListener('change', handleChange)
-                    updatePermission(permissionStatus.state)
-                })
-                .catch(() => {
-                    // for some user agents which implemented `query` method
-                    // but rise an error if specific permission name dose not supported
-                    updatePermission('granted')
+                    setState(permissionStatus.state)
                 })
+                // for some user agents which implemented `query` method
+                // but rise an error if specific permission name dose not supported
+                .catch(() => setState('granted'))
         } else if (checkPermissionApiUsability('request')) {
             navigator.permissions
                 .request({ name })
-                .then((p) => {
-                    updatePermission(p.state)
-                })
-                .catch(() => {
-                    updatePermission('granted')
-                })
+                .then((p) => setState(p.state))
+                .catch(() => setState('granted'))
         } else {
-            updatePermission('granted')
+            setState('granted')
         }
         return () => permissionStatus?.removeEventListener('change', handleChange)
-    }, [name, needRequest, permission])
-    return permission
+    })
```

Comment:
You should keep the original dependency array, if you omit it, the callback passed to `useEffect` will be executed every `componentDidUpdate`.